tests/ui/rfcs/rfc-2632-const-trait-impl/call-generic-method-nonconst.stderr

```diff
@@ -1,8 +1,8 @@
-error[E0277]: the trait bound `S: ~const Foo` is not satisfied
+error[E0277]: the trait bound `S: const Foo` is not satisfied
   --> $DIR/call-generic-method-nonconst.rs:23:34
    |
 LL | pub const EQ: bool = equals_self(&S);
-   |                      ----------- ^^ the trait `~const Foo` is not implemented for `S`
+   |                      ----------- ^^ the trait `const Foo` is not implemented for `S`
```

Comment:
This regression is a bit unfortunate. Of course, from the perspective of the lowered code, it makes sense that we register always-const obligations for const items, so technically speaking it's correct.

Still, it's bad that we basically say *`S: const Trait` is unsatisfied // required by bound `~const Trait` in `equals_self`*.